mars/services/scheduling/supervisor/manager.py

```diff
@@ -129,18 +255,47 @@ async def _get_execution_ref(self, band: BandType):
 
         return await mo.actor_ref(SubtaskExecutionActor.default_uid(), address=band[0])
 
-    async def finish_subtasks(self, subtask_ids: List[str], schedule_next: bool = True):
+    async def finish_subtasks(self,
+                              subtask_ids: List[str],
+                              bands: List[BandType] = None,
+                              schedule_next: bool = True):
+        logger.debug('Finished subtasks %s.', subtask_ids)
         band_tasks = defaultdict(lambda: 0)
-        for subtask_id in subtask_ids:
-            subtask_info = self._subtask_infos.pop(subtask_id, None)
+        bands = bands or [None] * len(subtask_ids)
+        for subtask_id, subtask_band in zip(subtask_ids, bands):
+            subtask_info = self._subtask_infos.get(subtask_id, None)
             if subtask_info is not None:
-                self._subtask_summaries[subtask_id] = subtask_info.to_summary(
-                    is_finished=True
-                )
+                self._subtask_summaries[subtask_id] = subtask_info.to_summary(is_finished=True)
+                subtask_info.end_time = time.time()
```

Comment:
Ditto. Making speculation strategy scattered in actors may not be a good practice.